pkg/route/api/v1/defaults.go

```diff
@@ -2,6 +2,18 @@ package v1
 
 import "k8s.io/kubernetes/pkg/runtime"
 
+func SetDefaults_RouteSpec(obj *RouteSpec) {
+	if len(obj.WildcardPolicy) == 0 {
+		obj.WildcardPolicy = "None"
+	}
+	switch obj.WildcardPolicy {
+	case WildcardPolicyType("None"):
+		obj.WildcardPolicy = WildcardPolicyNone
+	case WildcardPolicyType("Subdomain"):
+		obj.WildcardPolicy = WildcardPolicySubdomain
+	}
+}
```

Comment:
Do we want to do something on the case where something else is encountered?  Log an error and default to None?  Log and ignore the route (with appropriate status)?  Or is that definitely impossible by some other mechanism I haven't spotted.